### ISSUE TITLE
Ownership of /var/lib/nginx prevents large HTTP requests, other standard functionality

### ISSUE THREAD

**Issue by jeredfloyd** [sclorg/nginx-container#234]
The 1.20 / Dockerfile.rhel8 recursively changes ownership of /var/lib/nginx to 1001:0 with mode 770, however the nginx application runs as user 998.  Due to this, the nginx application is unable to write to /var/lib/nginx/tmp/client_body/ directory and large requests to the server result in 500 Internal Server Error.  This applies to other tmp directories used by nginx.

Example log errors:
  [crit] 7#0: *29 open() "/var/lib/nginx/tmp/client_body/0000000001" failed (13: Permission denied)
  [crit] 7#0: *56 open() "/var/lib/nginx/tmp/proxy/2/00/0000000002" failed (13: Permission denied)

For standalone (direct) container usage this can be resolved by adding:
`user default root;`
to the beginning of /etc/nginx/nginx.conf, but this may not be suitable for Kubernetes environments.

**Comment by zmiklank** [comment]
Hello and thank you for creating the issue.

I see the problem you are describing, I will try to summarize it again:

---

1) Running nginx as root by default creates two processes - one owned by root and one owned as user nginx:
```
[root@vm ~]# nginx
[root@vm ~]# ps aux | grep nginx
root        1360  0.0  0.0  13776  1352 ?        Ss   09:44   0:00 nginx: master process nginx
nginx       1361  0.0  0.2  14596  4520 ?        S    09:44   0:00 nginx: worker process
```
The worker in this case does not have access to the nginx-related files which were manipulated by in the Dockerfile, as they have 770 rights, and are owned by 1001:0:
```
NGINX_CONF_PATH=/etc/nginx/nginx.conf
NGINX_CONTAINER_SCRIPTS_PATH=/usr/share/container-scripts/nginx
NGINX_APP_ROOT=/opt/app-root
${NGINX_CONF_PATH}
${NGINX_APP_ROOT}/etc
${NGINX_APP_ROOT}/src/nginx-start/ 
${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start 
/var/lib/nginx /var/log/nginx /run
```

2) Running nginx from a standard user does not work out of the box (`mkdir() "/var/lib/nginx/tmp/client_body" failed (13: Permission denied)`). Therefore we change permissions and ownership of above-mentioned files and dirs in the Dockerfiles. And the port can be mapped when connecting to container.

Then running nginx as a non-root user lead to both processes being ran under the user:
```
[user@vm ~]$ nginx
[user@vm ~]$ ps aux | grep nginx
user     1495  0.0  0.0  13776  1352 ?        Ss   09:58   0:00 nginx: master process nginx
user     1496  0.0  0.1  14204  2484 ?        S    09:58   0:00 nginx: worker process
```

----

We need to support both (1) and 2)) use cases in this repository.
I think that we also need to preserve the GID=0, because of the openshift usecase.


Options:
1) add nginx user to root group - not really secure, I consider this not such good solution.
2) find out if we can use `user default root;`, as @jeredfloyd suggested.
3) ?

@uhliarik @notroj @pkubatrh  Do you have any ideas or observations?

**Comment by jeredfloyd** [comment]
Yes, this is a correct understanding of my issue.

However, I have just updated to tag 1-84 (updated 1 day ago) from the container catalog, and now the nginx master process runs as the default user!  It looks like this report was a duplicate of #231 which was resolved in #232 , so I am closing this issue.
